### PR TITLE
Fix shallow copy: 1.18.1

### LINKED ISSUE #5570: Shallow Git Clone + Checkout = git error: pathspec
The latest version (`conan 1.18.0`) introduces a regression in my codebase.
In particular, the commit [ab2d738bb1fc37a3ac4ec7e727d3df49fd3d4015](https://github.com/conan-io/conan/commit/ab2d738bb1fc37a3ac4ec7e727d3df49fd3d4015?diff=split) which introduces the *shallow git clone* feature.

When the package is built from source the shallow mode executes the following command:
~~~
scm.py: 92     self.repo.clone(..., shallow=True)
~~~
which executes the shallow clone:
~~~
git -c http.sslVerify=false init                                                                                                                                                                                                       
git -c http.sslVerify=false remote add origin "https://github.com/google/benchmark.git"                                                                                                                                                 
git -c http.sslVerify=false fetch --depth 1 origin "v1.4.1"                                                                                                                                                                             
git -c http.sslVerify=false checkout FETCH_HEAD
~~~

after it tries the checkout:
~~~
scm.py: 98     output += self.repo.checkout(element="v1.4.1", ...)
~~~

which produces the git error because the pathspec (v1.4.1.) does not exist:

~~~
error: pathspec 'v1.4.1' did not match any file(s) known to git 
~~~

Why do we need to checkout in the shallow copy? Wha I am missing?

### PR DESCRIPTION
**Pull request by lasote**
Changelog: Bugfix: The `scm` feature was trying to run a checkout after a shallow clone.
Docs: omit

Closes #5570 

@TAGS: svn, slow
@REVISIONS: 1

### CODE REVIEW

File: conans/test/functional/scm/scm_test.py [hunk 1]
```diff
@@ -290,6 +291,41 @@ def source(self):
...
+    def test_shallow_clone_remote(self, remote, revision, is_tag):
+        # https://github.com/conan-io/conan/issues/5570
+        self.client = TestClient()
+
+        # "remote" git
+        if remote == "local":
+            remote, _ = create_local_git_repo(tags=[revision] if is_tag else None,
+                                              files={"conans/model/username.py": "foo"})
+            if revision is None:  # Get the generated commit
+                revision = Git(remote).get_commit()
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: The commit is returned in the `_` of your previous call

File: conans/test/functional/scm/scm_test.py [hunk 2]
```diff
@@ -290,6 +291,41 @@ def source(self):
...
+class ConanLib(ConanFile):
+    name = "lib"
+    version = "0.1"
+    scm = {"type": "git", "url": "%s", "revision": "%s"}
+
+    def build(self):
+        assert os.path.exists(os.path.join(self.build_folder, "conans", "model", "username.py"))
+''' % (remote, revision)
+        self.client.save({"conanfile.py": conanfile})
+        create_local_git_repo(folder=self.client.current_folder, tags=[revision])
```
Comment (jgsogo) [inline conans/test/functional/scm/scm_test.py]: What is this second repo for?

File: conans/client/tools/scm.py [hunk 3]
```diff
@@ -145,23 +145,30 @@ def clone(self, url, branch=None, args="", shallow=False):
         return output
 
     def checkout(self, element, submodule=None):
+        # Element can be a tag, branch or commit
         self.check_repo()
         output = self.run('checkout "%s"' % element)
+        self.checkout_submodules(submodule)
```
Comment (jgsogo) [inline conans/client/tools/scm.py]:
```suggestion
        output += self.checkout_submodules(submodule)
```

File: conans/test/functional/scm/scm_test.py (hunk 2 above)
Comment (lasote) [reply conans/test/functional/scm/scm_test.py]: for nothing. removing it.

**Pull request merged**